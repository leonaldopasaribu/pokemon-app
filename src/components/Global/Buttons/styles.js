import styled from "@emotion/styled";

export const Button = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 30px auto 0 auto;
  font-size: 18px;
  font-style: normal;
  font-weight: 600;
  width: 100%;
  height: 60px;
  border: none;
  border-radius: 10px;
  border: ${(props) => (props.disabled ? "2px solid #14fc33" : "#14fc33")};
  background: ${(props) => (props.disabled ? "#fff" : "#14fc33")};
  color: ${(props) => (props.disabled ? "#14fc33" : "#fff")};
  cursor: ${(props) => (props.disabled ? "not-allowed" : "pointer")};

  &:hover {
    opacity: 0.8;
    transition: 0.3s;
  }

  @media only screen and (max-width: 600px) {
    font-size: 12px;
    height: auto;
    padding: 10px;
  }
`;
