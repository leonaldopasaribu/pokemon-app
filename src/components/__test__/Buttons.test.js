import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

import Buttons from "../Global/Buttons";

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Buttons></Buttons>, div);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<Buttons title="Previous"></Buttons>);
  expect(getByTestId("button")).toHaveTextContent("Previous");
});

it("matches snapshot Previous", () => {
  const tree = renderer.create(<Buttons label="Previous"></Buttons>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot Next", () => {
  const tree = renderer.create(<Buttons label="Next"></Buttons>).toJSON();
  expect(tree).toMatchSnapshot();
});
