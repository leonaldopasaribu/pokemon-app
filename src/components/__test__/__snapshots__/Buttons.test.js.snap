// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="css-j8msyy"
  data-testid="button"
/>
`;

exports[`matches snapshot Next 1`] = `
<button
  className="css-j8msyy"
  data-testid="button"
/>
`;

exports[`matches snapshot Previous 1`] = `
<button
  className="css-j8msyy"
  data-testid="button"
/>
`;
